spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/datastore/GcpDatastoreAutoConfiguration.java

```diff
@@ -101,21 +106,34 @@
 	}
 
 	@Bean
-	@ConditionalOnMissingBean
-	public Datastore datastore() {
-		DatastoreOptions.Builder builder = DatastoreOptions.newBuilder()
-				.setProjectId(this.projectId)
-				.setHeaderProvider(new UserAgentHeaderProvider(this.getClass()))
-				.setCredentials(this.credentials);
-		if (this.namespace != null) {
-			builder.setNamespace(this.namespace);
-		}
-
-		if (this.host != null) {
-			builder.setHost(this.host);
+	@ConditionalOnMissingBean(value = Datastore.class, parameterizedContainer = Supplier.class)
+	public Supplier<Datastore> datastoreSupplier(
+			ObjectProvider<org.springframework.cloud.gcp.autoconfigure.datastore.DatastoreNamespaceProvider> namespaceProvider,
```

Comment:
why the fully qualified name? is there another DatastoreNamespaceProvider?

Reply:
fixed!